native/acter/src/api/auth.rs

```diff
@@ -432,15 +445,50 @@ pub async fn register_with_token_under_config(
 
 impl Client {
     pub async fn deactivate(&self, password: String) -> Result<bool> {
-        // ToDo: make this a proper User-Interactive Flow rather than hardcoded for
-        //       password-only instance.
         let account = self.account()?;
         RUNTIME
             .spawn(async move {
-                let auth_data =
-                    AuthData::Password(Password::new(account.user_id().into(), password.clone()));
-                account.deactivate(None, Some(auth_data)).await?;
-                // FIXME: remove local data, too!
+                if let Err(e) = account.deactivate(None, None).await {
+                    let Some(inf) = e.as_uiaa_response() else {
+                        bail!("Server did not indicate how to allow deactivation.")
+                    };
+                    if let Some(err) = &inf.auth_error {
+                        bail!("Found auth error: {:?}", err.message);
+                    }
+                    let pswd = assign!(Password::new(account.user_id().into(), password), {
+                        session: inf.session.clone(),
+                    });
+                    let auth_data = AuthData::Password(pswd);
+                    account.deactivate(None, Some(auth_data)).await?;
+                    // FIXME: remove local data, too!
+                }
+                Ok(true)
+            })
+            .await?
+    }
+
+    pub async fn change_password(&self, old_val: String, new_val: String) -> Result<bool> {
+        let client = self.core.client().clone();
+        let account = self.account()?;
+        RUNTIME
+            .spawn(async move {
+                let capabilities = client.get_capabilities().await?;
+                if !capabilities.change_password.enabled {
+                    bail!("This client doesn't support password change");
+                }
+                if let Err(e) = account.change_password(&new_val, None).await {
+                    let Some(inf) = e.as_uiaa_response() else {
+                        bail!("Server did not indicate how to allow password change.")
+                    };
+                    if let Some(err) = &inf.auth_error {
+                        bail!("Found auth error: {:?}", err.message);
+                    }
```

Comment:
please use the new facilities to handle the error to show more understandable user errors.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1761/commits/4f1ee0e0417fc7e1ecc0119ff65aa79e6c6f8461